### PR TITLE
[script] [sell-loot] add hometown argument

### PR DESCRIPTION
**Pull request by KatoakDR**
### Dependencies
* https://github.com/rpherbig/dr-scripts/pull/5256

### Background
* Feature requested by [Tirost](https://discord.com/channels/745675889622384681/912127186419482664/913661837462089788)

> I'm requesting being able to use `;sell-loot crossing` or `;sell-loot hib` and have the script take me to that hometown's gem appraiser, tanner and bank. My hope is to be able to use `sell-loot` in locations other than the hometown I use for training to sell off invasion spoils, money of a denomination I'd rather not pay exchange fees on or that's been acquired in an RP event, or return money after shopping or trading.

### Changes
* Add `hometown` as a new script argument to override that value from character's YAML

### Example Usage
_Original two options, and these still exist_
```
;sell-loot
;sell-loot 10 silver
```
_New options allow you to specify the hometown to use_
```
;sell-loot riverhaven
;sell-loot shard 5 gold
```

### CODE REVIEW

File: sell-loot.lic [hunk 1]
```diff
@@ -60,23 +78,6 @@ class SellLoot
...
-      []
-    ]
-
-    args = parse_args(arg_definitions)
-
-    keep_amount = args.amount || setting[0] || 3
-    keep_type = args.type || setting[1] || 'silver'
-
-    convert_to_copper(keep_amount, keep_type)
-  end
```
Comment (KatoakDR) [inline sell-loot.lic:78]: Parsing of arguments and determining how much coin to keep on your person is now part of the `initialize` method

File: sell-loot.lic [hunk 2]
```diff
@@ -19,16 +36,19 @@ class SellLoot
 
     @settings = get_settings
     town_data = get_data('town')
-    @character_hometown = @settings.hometown
-    @hometown = town_data[@settings.hometown]
+    @character_hometown = DRC.get_town_name(args.hometown || @settings.hometown)
+    @hometown = town_data[@character_hometown]
```
Comment (KatoakDR) [inline sell-loot.lic:40]: Uses the new `DRC.get_town_name(..)` method introduced by https://github.com/rpherbig/dr-scripts/pull/5256

File: sell-loot.lic [hunk 3]
```diff
@@ -19,16 +36,19 @@ class SellLoot
...
     @bankbot_room_id = @settings.bankbot_room_id
     @bankbot_deposit_threshold = @settings.bankbot_deposit_threshold
     @bankbot_enabled = @settings.bankbot_enabled
     @local_currency = town_data[@character_hometown]['currency']
     skip_bank = @settings.sell_loot_skip_bank
     skip_exchange = @settings.sell_loot_skip_exchange
-    keep_money_by_currency = @settings.sell_loot_money_on_hand
+    keep_money_by_currency = @settings.sell_loot_money_on_hand.split(' ')
+    keep_amount = args.amount || keep_money_by_currency[0] || 3
+    keep_denomination = args.type || keep_money_by_currency[1] || 'silver'
```
Comment (KatoakDR) [inline sell-loot.lic:50]: The defaults of 3 silver was original logic; just moved here from the now-deleted `parse(..)` method

**Comment by rpherbig** [comment]
Code changes look good ✔️

**Comment by KatoakDR** [comment]
This is ready, thanks

**Pull request merged**